src/artm/core/collection_parser.cc

```diff
@@ -24,6 +24,26 @@
 
 using boost::iostreams::mapped_file_source;
 
+namespace {
+class ifstream_or_cin {
```

Comment:
Hey, what the difference between `ifstream` and `cin`?
It seemed to me that `cin` is a special instance of `ifstream` (or maybe just `istream`). In unix systems there are a special filenames for cin/cout/cerr: `/dev/stdin`, ...